x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypes.java

```diff
@@ -212,7 +219,8 @@ public static boolean isRepresentable(DataType t) {
             && t != FLOAT
             && t != SCALED_FLOAT
             && t != SOURCE
-            && t != HALF_FLOAT;
+            && t != HALF_FLOAT
+            && isCounterType(t) == false;
```

Comment:
I think counters are representable though. They can be loaded into blocks now.

Reply:
I disabled this mainly to avoid the map in AbstractFunctionTestCase https://github.com/elastic/elasticsearch/pull/107877#discussion_r1579527982. Would you be okay if I enable this in a follow-up?

Reply:
:+1:

Delaying is fine.